### PR TITLE
article dependent destroying

### PR DESCRIPTION
**Pull request by jkao1**
With dependent destroying, deleting an article will delete its:
- outquotes
- authorships
- comments
- media

Deleting a user will delete its:
- authorships
- comments
- UserRoles

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@ class Article < ApplicationRecord
   extend FriendlyId
   friendly_id :title, use: :slugged
   belongs_to :section, optional: true
-  has_many :users, through: :authorships
+  has_many :authorships
```
Comment (NicholasLYang) [inline app/models/article.rb]: Why this?

File: app/models/article.rb [hunk 2]
```diff
@@ -2,8 +2,9 @@ class Article < ApplicationRecord
...
+  has_many :users, through: :authorships, dependent: :destroy
   has_many :media
```
Comment (NicholasLYang) [inline app/models/article.rb]: Probably should delete media too, no?

File: app/models/user.rb [hunk 3]
```diff
@@ -4,7 +4,8 @@ class User < ApplicationRecord
           :recoverable, :rememberable, :trackable, :validatable,
           :confirmable, :omniauthable
   include DeviseTokenAuth::Concerns::User
-  has_many :articles, through: :authorships
+  has_many :authorships
+  has_many :articles, through: :authorships, dependent: :destroy
   has_many :comments
```
Comment (NicholasLYang) [inline app/models/user.rb]: Do we want to preserve comments? If so we need to make sure it isn't an error to have a comment with no author

File: app/models/user.rb [hunk 4]
```diff
@@ -4,7 +4,8 @@ class User < ApplicationRecord
...
   has_many :user_roles
   has_many :roles, through: :user_roles
```
Comment (NicholasLYang) [inline app/models/user.rb]: Should also be `dependent: :destroy`

File: app/models/article.rb (hunk 1 above)
Comment (jkao1) [reply app/models/article.rb]: Removal of that extra `has_many :authorships` results in a 500 error:
```
ActiveRecord::HasManyThroughAssociationNotFoundError (Could not find the association :authorships in model Article):
```

**Pull request merged**